<launch>
    <arg name="cmd_src" default="teleop"/>
    <arg name="robot" default="nusim"/>
  
    <node pkg="tf2_ros"
        exec="static_transform_publisher"
        args="0 0 0 0 0 0 nusim/world map"/>
    <node pkg="rviz2" exec="rviz2" args="-d $(find-pkg-share nuslam)/config/slam.rviz"/> 
  
    <include file="$(find-pkg-share nuturtle_description)/launch/load_one.launch.py" if="$(eval '\'$(var robot)\' == \'nusim\'')" >
      <arg name="use_rviz" value="false"/>
      <arg name="use_jsp" value="true"/>
      <arg name="color" value="green"/>
    </include>
    <include file="$(find-pkg-share nuturtle_control)/launch/start_robot.launch.xml">
      <arg name="cmd_src" value="$(var cmd_src)"/>
      <arg name="robot" value="$(var robot)"/>
      <arg name="use_rviz" value="false"/>
    </include>
  
    <node pkg="nuslam" exec="slam" name="slam" if="$(eval '\'$(var robot)\' == \'nusim\'')" >
      <param from="$(find-pkg-share nuturtle_description)/config/diff_params.yaml"/>
      <param name="body_id" value="green/base_footprint"/>
      <param name="odom_id" value="green/odom"/>
      <param name="wheel_left" value="wheel_left_joint"/>
      <param name="wheel_right" value="wheel_right_joint"/>
      <remap from="joint_states" to="blue/joint_states"/>
    </node>
  
  </launch>